tests/stat.rs

```diff
@@ -0,0 +1,43 @@
+extern crate nix;
+
+#[cfg(test)]
+mod test {
+    use nix::sys::stat::stat;
+
+
+    #[test]
+    fn test_stat() {
+
+        use nix::fcntl::open;
+        use nix::unistd::{close, unlink};
+        use nix::fcntl::O_CREAT;
+        use nix::sys::stat::S_IWUSR;
+
+        let filename = b"target/foo.txt";
+        let fd_res = open(filename, O_CREAT, S_IWUSR);
+        assert!(fd_res.is_ok()); // creating a file in "target" should always work
+        let close_res = close(fd_res.ok().unwrap()); // close right here. We use the file only for the stat test
+        assert!(close_res.is_ok());
```

Comment:
You can also use `Result::unwrap()` here. It'll do the unwrap + assert in one step and even print the `Err` value if it's an `Err`.

```rust
let fd_res = open(...).unwrap();
let close_res = close(fd_res).unwrap();
```